pkg/ebpf/telemetry/docs/errors_telemetry.md

```diff
@@ -0,0 +1,56 @@
+# EBPF errors telemetry
+EBPF errors telemetry collection refers to collecting error codes for failed helper operations. We collect two types of failures:
+- Map update failures.
+- Helper failures. These include:
+    - `bpf_probe_read`
+    - `bpf_probe_read_user`
+    - `bpf_probe_read_kernel`
+    - `bpf_skb_load_bytes`
+    - `bpf_perf_event_output`
+
+For maps, errors are collected per map per error code.
+```json
+{
+    "ebpf_maps__errors":{
+      "conn_stats":{
+         "EEXISTS":3787
+      },
+      "http_in_flight":{
+         "EEXISTS":35
+      },
+      "tcp_stats":{
+         "EEXISTS":3778
+      }
+   }
+}
+```
+For helpers, errors are collected per probe per helper per error code.
+```json
+{
+"ebpf_helpers__errors":{
+      "kretprobe__do_sys_openat2":{
+         "bpf_probe_read_user":{
+            "EFAULT":8
+         }
+      }
+   }
+}
+```
+
+## Implementation
+This telemetry collection mechanism has two parts
+1. Telemetry instrumentation
```

Comment:
I find the wording "Telemetry instrumentation" confusing. We are instrumenting the helper functions. The telemetry is the data we collect as a result of that. As written, it seems like we are instrumenting some existing telemetry.

Reply:
> We are instrumenting the helper functions
The instrumentation refers to the trampoline instruction jumping to the end of the bytecode. It is separate from the helper function macros.

Reply:
Could we call that a preamble or something? I think that makes it clearer what is happening. "Telemetry instrumentation" is confusing to me, it sounds like we are instrumenting telemetry.